### PR TITLE
Configuration along the maneuver can be queried

### PR DESCRIPTION
**Pull request by slamajakub**
Methods for querying vehicle configuration along the path added.

### CODE REVIEW

File: src/AcceleratedDubins.jl [hunk 1]
```diff
@@ -3,7 +3,7 @@ module AcceleratedDubins
 using Optim
 
 export
-    path_len, path_time, sample_path, speed_by_radius, speed_profile, fastest_path, retrieve_path, optimized_path, possible_tuples, radii_samples_lin, radii_samples_exp
+    path_len, path_time, sample_path, speed_by_radius, speed_profile, fastest_path, retrieve_path, optimized_path, possible_tuples, radii_samples_lin, radii_samples_exp, get_configuration
```
Comment (petvana) [inline src/AcceleratedDubins.jl]:
```suggestion
    path_len, path_time, sample_path, speed_by_radius, speed_profile, fastest_path, retrieve_path, optimized_path, 
    possible_tuples, radii_samples_lin, radii_samples_exp, get_configuration
```

File: src/AcceleratedDubins.jl [hunk 2]
```diff
@@ -501,63 +449,43 @@ Sample single path for plotting
...
+
+        # Add last point of first arch
+        x, y, theta = get_configuration(path, path.lengths[1])
+        push!(confx, x)
+        push!(confy, y)
+
+        # Sample line segment
+        x, y = point_by_angle([x, y], theta, path.lengths[2])       
+        push!(confx, x)
+        push!(confy, y)
```
Comment (slamajakub) [inline src/AcceleratedDubins.jl:471]: As originally, straight segment is sampled as two points. If this is not desirable anymore, whole function could be reduced to a single for-loop, similarly to first arc sampling

File: src/AcceleratedDubins.jl [hunk 3]
```diff
@@ -433,58 +434,6 @@ end
...
-- `start::Vector{Float64}`: vector of starting point: [x, y, _] or [x, y]
-- `a1::Float64`: starting angle in radians
-- `a2::Float64`: ending angle in radians
-- `r::Float64`: radius of the arc
-- `part::Float64`: how smoothly to sample the arc, smaller -> finer
-- `dir::Int`: direction of the arc, 1 -> clockwise, -1 -> counter-clockwise
-
-# Return
-- `confx::Vector{Float64}`: x points of the path to plot
-- `confy::Vector{Float64}`: y points of the path to plot
```
Comment (petvana) [inline src/AcceleratedDubins.jl:452]: @slamajakub  Can you use and update this comment for the new function, please.

File: src/AcceleratedDubins.jl [hunk 4]
```diff
@@ -674,7 +603,137 @@ function LSR_path(alfa, beta, dist, radius1, radius2)
...
+# Return
+- `ret::Vector{Float64}`: vehicle configuration [x, y, theta]
+"""
+function get_configuration(path::DubinsPathR2, distance::Float64)
+    # Clip distance at end of the maneuver
+    @assert distance >= 0. "Distance where configuration is queried must be greater than 0.0"
+    if distance > sum(path.lengths)
+        distance = sum(path.lengths)
+    end
+
```
Comment (petvana) [inline src/AcceleratedDubins.jl:650]:
```suggestion
```

File: src/AcceleratedDubins.jl [hunk 5]
```diff
@@ -674,7 +603,137 @@ function LSR_path(alfa, beta, dist, radius1, radius2)
...
+            ang2 = ang1 - distance/path.r[1]
+            r = path.r[1]
+
+            x, y, theta = get_configuration_in_turn(path.origin, ang1, ang2, r, R_SEG)
+        end
+    elseif distance <= path.lengths[1] + path.lengths[2]
+        # Requested conf is in second segment
+        x = nothing
+        y = nothing
+        theta = nothing
```
Comment (petvana) [inline src/AcceleratedDubins.jl]:
```suggestion
        x = y = theta = nothing
```

File: src/AcceleratedDubins.jl [hunk 6]
```diff
@@ -674,7 +603,137 @@ function LSR_path(alfa, beta, dist, radius1, radius2)
...
+            ang2 = ang1 + distance/path.r[1]
+            r = path.r[1]
+
+            x, y, theta = get_configuration_in_turn(path.origin, ang1, ang2, r, L_SEG)
+        else
+            ang1 = path.origin[3]+pi/2
+            ang2 = ang1 - distance/path.r[1]
+            r = path.r[1]
+
+            x, y, theta = get_configuration_in_turn(path.origin, ang1, ang2, r, R_SEG)
```
Comment (petvana) [inline src/AcceleratedDubins.jl]:
```suggestion
            r = path.r[1]
            x, y, theta = get_configuration_in_turn(path.origin, ang1, ang2, r, L_SEG)
        else
            ang1 = path.origin[3]+pi/2
            ang2 = ang1 - distance/path.r[1]
            r = path.r[1]
            x, y, theta = get_configuration_in_turn(path.origin, ang1, ang2, r, R_SEG)
```

File: src/AcceleratedDubins.jl [hunk 7]
```diff
@@ -674,7 +603,137 @@ function LSR_path(alfa, beta, dist, radius1, radius2)
...
+
+            x, y, theta = get_configuration_in_turn(path.origin, ang1, ang2, r, L_SEG)
+        else
+            ang1 = path.origin[3]+pi/2
+            ang2 = ang1 - path.lengths[1]/path.r[1]
+
+            r = path.r[1]
+
+            x, y, theta = get_configuration_in_turn(path.origin, ang1, ang2, r, R_SEG)
+        end
```
Comment (petvana) [inline src/AcceleratedDubins.jl]:
```suggestion
            ang2 = ang1 + path.lengths[1]/path.r[1]
            r = path.r[1]
            x, y, theta = get_configuration_in_turn(path.origin, ang1, ang2, r, L_SEG)
        else
            ang1 = path.origin[3]+pi/2
            ang2 = ang1 - path.lengths[1]/path.r[1]
            r = path.r[1]
            x, y, theta = get_configuration_in_turn(path.origin, ang1, ang2, r, R_SEG)
        end
```

File: src/AcceleratedDubins.jl [hunk 8]
```diff
@@ -674,7 +603,137 @@ function LSR_path(alfa, beta, dist, radius1, radius2)
...
+            ang2 = ang1 + path.lengths[1]/path.r[1]
+            r = path.r[1]
+
+            x, y, theta = get_configuration_in_turn(path.origin, ang1, ang2, r, L_SEG)
+        else
+            ang1 = path.origin[3]+pi/2
+            ang2 = ang1 - path.lengths[1]/path.r[1]
+            r = path.r[1]
+
+            x, y, theta = get_configuration_in_turn(path.origin, ang1, ang2, r, R_SEG)
```
Comment (petvana) [inline src/AcceleratedDubins.jl]:
```suggestion
            r = path.r[1]
            x, y, theta = get_configuration_in_turn(path.origin, ang1, ang2, r, L_SEG)
        else
            ang1 = path.origin[3]+pi/2
            ang2 = ang1 - path.lengths[1]/path.r[1]
            r = path.r[1]
            x, y, theta = get_configuration_in_turn(path.origin, ang1, ang2, r, R_SEG)
```

File: src/AcceleratedDubins.jl [hunk 9]
```diff
@@ -674,7 +603,137 @@ function LSR_path(alfa, beta, dist, radius1, radius2)
...
+        x, y = point_by_angle([x, y], theta, path.lengths[2])
+        distance -= (path.lengths[1] + path.lengths[2])
+
+        # Find the configuration within the third segment
+        if Integer(path.type) == Integer(LSL) || Integer(path.type) == Integer(RSL)
+            ang1 = theta-pi/2
+            ang2 = ang1 + distance/path.r[3]
+            r = path.r[3]
+
+            x, y, theta = get_configuration_in_turn([x, y], ang1, ang2, r, L_SEG)
```
Comment (petvana) [inline src/AcceleratedDubins.jl]:
```suggestion
            r = path.r[3]
            x, y, theta = get_configuration_in_turn([x, y], ang1, ang2, r, L_SEG)
```

File: src/AcceleratedDubins.jl [hunk 10]
```diff
@@ -674,7 +603,137 @@ function LSR_path(alfa, beta, dist, radius1, radius2)
...
+        else
+            ang1 = theta+pi/2
+            ang2 = ang1 - distance/path.r[3]
+            r = path.r[3]
+
+            x, y, theta = get_configuration_in_turn([x, y], ang1, ang2, r, R_SEG)
```
Comment (petvana) [inline src/AcceleratedDubins.jl]:
```suggestion
            r = path.r[3]
            x, y, theta = get_configuration_in_turn([x, y], ang1, ang2, r, R_SEG)
```

File: src/AcceleratedDubins.jl [hunk 11]
```diff
@@ -674,7 +603,124 @@ function LSR_path(alfa, beta, dist, radius1, radius2)
...
+        end
+
+        # Find configuration within the second segment
+        distance -= path.lengths[1]
+        x, y = point_by_angle([x, y], theta, distance)
+    else
+        # Requested configuration is in the third segment
+        x = nothing
+        y = nothing
+        theta = nothing
```
Comment (petvana) [inline src/AcceleratedDubins.jl]:
```suggestion
        x = y = theta= nothing
```

File: src/AcceleratedDubins.jl [hunk 12]
```diff
@@ -500,63 +449,43 @@ Sample single path for plotting
 function sample_path(path::DubinsPathR2, resolution::Float64 = 0.01)
     if path == nothing
         return [], []
```
Comment (petvana) [inline src/AcceleratedDubins.jl]: This returns type `Any[]`, thus the return type of the `sample_path` may contain type instability.

Comment (slamajakub) [reply src/AcceleratedDubins.jl]: Remained in the code from original implementation. I'll fix in the code.
```suggestion
        return Vector{Float64}(), Vector{Float64}()
```

**Pull request merged**